actool/manifest.go

```diff
@@ -0,0 +1,364 @@
+		stderr("patch-manifest: --manifest is incompatible with other manifest editing options")
+		return 1
+	}
+
+	inputFile = args[0]
+
+	// Prepare output writer
+
+	if patchReplace {
+		fh, err = ioutil.TempFile(path.Dir(inputFile), ".actool-tmp."+path.Base(inputFile)+"-")
+		if err != nil {
+			stderr("patch-manifest: Cannot create temporary file: %v", err)
+			return 1
+		}
+	} else {
+		outputFile = args[1]
+
+		ext := filepath.Ext(outputFile)
+		if ext != schema.ACIExtension {
+			stderr("patch-manifest: Extension must be %s (given %s)", schema.ACIExtension, ext)
+			return 1
+		}
+
+		mode := os.O_CREATE | os.O_WRONLY
+		if patchOverwrite {
+			mode |= os.O_TRUNC
+		} else {
+			mode |= os.O_EXCL
+		}
+
+		fh, err = os.OpenFile(outputFile, mode, 0644)
+		if err != nil {
+			if os.IsExist(err) {
+				stderr("patch-manifest: Output file exists (try --overwrite)")
+			} else {
+				stderr("patch-manifest: Unable to open output %s: %v", outputFile, err)
+			}
+			return 1
+		}
+	}
+
+	var gw *gzip.Writer
+	var w io.WriteCloser = fh
+	if !patchNocompress {
+		gw = gzip.NewWriter(fh)
+		w = gw
+	}
+	tw := tar.NewWriter(w)
+
+	defer func() {
+		tw.Close()
+		if !patchNocompress {
+			gw.Close()
+		}
+		fh.Close()
+		if exit != 0 && !patchOverwrite {
+			os.Remove(fh.Name())
+		}
+	}()
+
+	// Prepare input reader
+
+	input, err := os.Open(inputFile)
+	if err != nil {
+		stderr("patch-manifest: cannot open %s: %v", inputFile, err)
+		return 1
+	}
+	defer input.Close()
+
+	tr, err := aci.NewCompressedTarReader(input)
+	if err != nil {
+		stderr("patch-manifest: cannot extract %s: %v", inputFile, err)
+		return 1
+	}
+
+	var newManifest []byte
+
+	if patchManifestFile != "" {
+		mr, err := os.Open(patchManifestFile)
+		if err != nil {
+			stderr("patch-manifest: cannot open %s: %v", patchManifestFile, err)
+			return 1
+		}
+		defer input.Close()
+
+		newManifest, err = ioutil.ReadAll(mr)
+		if err != nil {
+			stderr("patch-manifest: cannot read %s: %v", patchManifestFile, err)
+			return 1
+		}
+	}
+
+	err = iterate(tr, tw, false, newManifest)
+	if err != nil {
+		stderr("patch-manifest: Unable to read %s: %v", inputFile, err)
+		return 1
+	}
+
+	if patchReplace {
+		err = os.Rename(fh.Name(), inputFile)
+		if err != nil {
+			stderr("patch-manifest: Cannot rename %q to %q: %v", fh.Name, inputFile, err)
+			return 1
+		}
+	}
+
+	return
+}
+
+func runCatManifest(args []string) (exit int) {
+	if len(args) != 1 {
+		stderr("cat-manifest: Must provide one file")
+		return 1
+	}
+
+	inputFile = args[0]
+
+	input, err := os.Open(inputFile)
+	if err != nil {
+		stderr("cat-manifest: cannot open %s: %v", inputFile, err)
+		return 1
+	}
+	defer input.Close()
+
+	tr, err := aci.NewCompressedTarReader(input)
+	if err != nil {
+		stderr("cat-manifest: cannot extract %s: %v", inputFile, err)
+		return 1
+	}
+
+	err = iterate(tr, nil, true, nil)
+	if err != nil {
+		stderr("cat-manifest: Unable to read %s: %v", inputFile, err)
```

Comment:
nit, inconsistent capitalisation, please standardise it across the output
`cat-manifest: cannot extract`
`cat-manifest: Unable to read`